modules/PanelApp.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+"""Evidence parser for the Genomics England PanelApp data."""
+
+import argparse
+import logging
+import os
+import re
+import tempfile
+
+import requests
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import (
+    array, array_distinct, col, collect_set, concat, explode, lit, regexp_extract, regexp_replace, split, trim, when
+)
+
+
+class PanelAppEvidenceGenerator:
+
+    # Fixes which are applied to original PanelApp phenotype strings *before* splitting them by semicolon.
+    PHENOTYPE_BEFORE_SPLIT_RE = {
+        # Fixes for specific records.
+        r'\(HP:0006574;\);': r'(HP:0006574);',
```

Comment:
Sometimes, the only way is to address the specific cases. For example, on the line highlighted in this thread, the source record looks like this: `(HP:0006574;);` — an extra semicolon interferes with phenotype splitting. Unfortunately there's no easy way to generalise this. I hope that once I report the list of issues to PanelApp, they can eventually clean this up and avoid similar things in the future, so that this becomes less reelvant.

We don't want to remove all round brackets in all cases, because sometimes they do convey meaningful grammar:
* `Peroxisome biogenesis disorder 7A (Zellweger)`
* `INFANTILE REFSUM DISEASE (IRD)`
* `Muscular dystrophy-dystroglycanopathy (congenital with mental retardation), type B, 6`

In contrast, the curly brackets are always removed, which is already implemented.